tests/test_flow/test_reloading.py

```diff
@@ -0,0 +1,277 @@
+import pytest
+
+import sys
+
+from bionic.exception import CodeVersioningError
+
+
+@pytest.fixture
+def tmp_module_path(tmp_path):
+    tmp_dir_name = str(tmp_path)
+    sys.path.insert(0, tmp_dir_name)
+
+    # Pytest rewrites modules and caches the compiled pyc file. The cache can
+    # interfere with the tests below. Disable the cache.
+    dont_write_bytecode = sys.dont_write_bytecode
+    sys.dont_write_bytecode = True
+
+    yield tmp_path
+
+    # Ensure that the temporary module is not loaded by a future test
+    sys.path.remove(tmp_dir_name)
+
+    sys.dont_write_bytecode = dont_write_bytecode
+
+
+@pytest.fixture
+def write_flow_file(tmp_module_path):
+    modules = set()
+    counter = 0
+
+    def _write_flow_file(module_name, contents):
+        nonlocal counter
+        counter += 1
+        (tmp_module_path / f"{module_name}.py").write_text(contents)
+        modules.add(module_name)
+
+    yield _write_flow_file
+
+    # Invalidate caches
+    for m in modules:
+        sys.modules.pop(m, None)
+
+
+@pytest.fixture
+def write_flow(write_flow_file):
+    def _write_flow(x):
+        write_flow_file(
+            "flow",
+            f"""
+import bionic as bn
+
+builder = bn.FlowBuilder('flow')
+builder.assign('x', {x})
+
+@builder
+def x_doubled(x):
+    return x*2
+
+flow = builder.build()
+        """,
+        )
+
+    return _write_flow
+
+
+def verify_flow_contents(flow, results_map):
+    assert {k: flow.get(k) for k in results_map.keys()} == results_map
+    assert {k: getattr(flow.get, k)() for k in results_map.keys()} == results_map
+
+
+@pytest.fixture(params=["reloading in place", "not reloading in place"])
```

Comment:
nit: You can also call these params `reload` and `reloading` to be concise.

Reply:
I can see that concern, but I also think it will be clear in the contexts where it matters.

Also, "not reloading in place" is a bit ambiguous -- it suggests to me that no reloading is happening at all. WDYT about `reload_in_place` and `reloading_copy`?

Reply:
Actually, on further thought, my concern can be resolved by renaming the pytest method to `reload_method`. Renamed the params to use `reload` and `reloading`